subsys/fs/Kconfig

```diff
@@ -39,6 +39,13 @@ config FILE_SYSTEM_NFFS
 	  Note: NFFS requires 1-byte unaligned access to flash thus it
 	  will not work on devices that support only aligned flash access.
 
+config FILE_SYSTEM_LITTLEFS
+	bool "LittleFS file system support"
+	depends on FLASH_MAP
+	depends on FLASH_PAGE_LAYOUT
```

Comment:
According to the samples littlefs depends on newlib, so I would expect a `depends on NEWLIB_LIBC` here

Reply:
littlefs depends on an implementation of `strspn` and `strcspn`.  newlib is not the only provider of that capability, it's just that minimal libc doesn't have it.

Probably the right answer is to implement those two functions for minimal libc, but that's not in scope right now.